por/sigchain.go

```diff
@@ -185,3 +187,36 @@ func VerifySigChainPath(sc *pb.SigChain, height uint32) error {
 
 	return nil
 }
+
+func SignatureHashWithPenalty(sc *pb.SigChain) ([]byte, error) {
+	blockHash, err := common.Uint256ParseFromBytes(sc.BlockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	if blockHash == common.EmptyUint256 {
+		return nil, errors.New("block hash in sigchain is empty")
+	}
+
+	height, err := store.GetHeightByBlockHash(blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	leftShiftBit := 0
+	if config.SigChainRecentMinerBitShift.GetValueAtHeight(height) > 0 && config.SigChainRecentMinerBlocks > 0 {
+		rm, err := GetRecentMiner(sc.BlockHash)
+		if err != nil {
+			return nil, err
+		}
+
+		count := 0
+		for i := 0; i < sc.Length()-2; i++ {
+			count += rm[string(sc.Elems[i].NextPubkey)]
```

Comment:
All keys of RecentMiner just only signerPk.
It means that there are only `config.SigChainRecentMinerBlocks` skipper check point. It is not enough maybe.

Reply:
That's the goal. We don't want add penalty for relayers that do not earn any reward.

Reply:
What if the key `string(sc.Elems[i].NextPubkey)` doesn't exist?

Reply:
The value will be 0